### ISSUE TITLE
quartus-prime-lite segfault on startup

### ISSUE THREAD

**Issue by Abaan404** [NixOS/nixpkgs#346368]
### Describe the bug
The packaged quartus segfaults on startup when trying to launch it (i.e. running `quartus-prime-lite` on the cli)

### Steps To Reproduce
1. Install and build quartus-prime-lite (via nix-shell or any prefered packaging method from nixpkgs)
2. run quartus-prime-lite from the cli. 
3. Nothing happens, no stdout and no quartus
4. run `journalctl` and find a coredump reported with the following stacktrace:
```
                                               Module libprotobuf.so.3.17.3.0 without build-id.
                                               Stack trace of thread 78149:
                                               #0  0x00007f317f4a2efc __pthread_kill_implementation (/nix/store/c10zhkbp6jmyh0xc5kd123ga8yy2p4hk-glibc-2.39->
                                               #1  0x00007f317f452e86 raise (/nix/store/c10zhkbp6jmyh0xc5kd123ga8yy2p4hk-glibc-2.39-52/lib/libc.so.6 + 0x3fe>
                                               #2  0x00007f317f43b935 abort (/nix/store/c10zhkbp6jmyh0xc5kd123ga8yy2p4hk-glibc-2.39-52/lib/libc.so.6 + 0x289>
                                               #3  0x00007f3180297957 _ZNK14QMessageLogger5fatalEPKcz (libQt5Core.so.5 + 0x97957)
                                               #4  0x00007f3180b0672b _ZN22QGuiApplicationPrivate25createPlatformIntegrationEv (libQt5Gui.so.5 + 0x10672b)
                                               #5  0x00007f3180b06ba0 _ZN22QGuiApplicationPrivate21createEventDispatcherEv (libQt5Gui.so.5 + 0x106ba0)
                                               #6  0x00007f31804d4856 _ZN23QCoreApplicationPrivate4initEv (libQt5Core.so.5 + 0x2d4856)
                                               #7  0x00007f3180b09c4f _ZN22QGuiApplicationPrivate4initEv (libQt5Gui.so.5 + 0x109c4f)
                                               #8  0x00007f3181368b29 _ZN19QApplicationPrivate4initEv (libQt5Widgets.so.5 + 0x168b29)
                                               #9  0x00007f319a0a9f99 _ZN8AFCQ_APPC2ERiPPc (libgcl_afcq.so + 0xa9f99)
                                               #10 0x00007f319b7814ba _ZN7QUI_APPC2ERiPPc (libsys_qui.so + 0x614ba)
                                               #11 0x00007f319b921b99 _ZN7QGQ_APPC2ERiPPc (libsys_qgq.so + 0x12b99)
                                               #12 0x0000564c43ffc048 _Z8qgq_mainiPPKc (quartus + 0x2048)
                                               #13 0x00007f318c4d95c0 _Z15msg_main_threadPv (libccl_msg.so + 0x3e5c0)
                                               #14 0x00007f318b164c9c thr_final_wrapper (libccl_thr.so + 0x5c9c)
                                               #15 0x00007f318c4d968a _Z18msg_thread_wrapperPFPvS_ES_ (libccl_msg.so + 0x3e68a)
                                               #16 0x00007f318c413096 _Z18mem_thread_wrapperPFPvS_ES_ (libccl_mem.so + 0xc096)
                                               #17 0x00007f318b6661a8 _Z18err_thread_wrapperPFPvS_ES_ (libccl_err.so + 0x91a8)
                                               #18 0x00007f318b164cdf thr_thread_wrapper (libccl_thr.so + 0x5cdf)
                                               #19 0x00007f318c4db58a _Z12msg_exe_mainiPPKcPFiiS1_E (libccl_msg.so + 0x4058a)
                                               #20 0x0000564c43ffc165 main (quartus + 0x2165)
                                               #21 0x00007f317f43d10e __libc_start_call_main (/nix/store/c10zhkbp6jmyh0xc5kd123ga8yy2p4hk-glibc-2.39-52/lib/>
                                               #22 0x00007f317f43d1c9 __libc_start_main@@GLIBC_2.34 (/nix/store/c10zhkbp6jmyh0xc5kd123ga8yy2p4hk-glibc-2.39->
                                               #23 0x0000564c43ffbf39 _start (quartus + 0x1f39)
                                               ELF object binary architecture: AMD x86-64
```

### Expected behavior
Installing the package and running it should run quartus

### Screenshots
N/A

### Additional context
Seems to be crashing in the bundled qt5 library. Tried running a debugger to extract the exact error message (since quartus seems to have it disabled?) but I couldnt figure it out. I'm uncertain if my env is the error or the package is at fault however since nixos claims th be reproducible I dont see why my env would break quartus from launching.

### Notify maintainers

@bjornfor 
@kwohlfahrt 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.52, NixOS, 24.05 (Uakari), 24.05.20240928.fbca5e7`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.8`
 - channels(root): `"home-manager-24.05.tar.gz, nixos-24.05"`
 - nixpkgs: `/nix/store/mrpfclqc430y8gc7xs3rrp59cjk1pc9b-source`
```

Thanks!

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by Abaan404** [comment]
running `quartus` gives the same error, looking at `quartus-prime-lite` it seems to be a wrapper shell script that runs `quartus` under bwrap, not sure what its doing.

Im under wayland and running a minimal config with a wm if that helps. Removing all qt related changes doesnt seem to do anything either not sure whats going on. I can set up a [temp repo](https://github.com/Abaan404/dotfiles/tree/27f8ba1993303a1460052f4808fd376fd4ebe9e1) temporarily if that helps (everything nix related is under /nix/) ive only started using nix since last weekend so its very likely I could have messed something up.

**Comment by bjornfor** [comment]
> it seems to be a wrapper shell script that runs quartus under bwrap, not sure what its doing.

It sets up the FHS env that allows running Quartus (which is built for FHS distros) on NixOS, which is a non-FHS distro. 

Here's a minimal repro that shows it _working_:

```nix
# file: test.nix
let
  nixpkgs = builtins.fetchTarball {
    url = "https://github.com/NixOS/nixpkgs/archive/eeeb90a1dd3c9bea3afdbc76fd34d0fb2a727c7a.tar.gz"; # nixpkgs-unstable @ 2024-10-04
    sha256 = "01c17c4y7jmjwp82djrap8hp7w2a46vy4sg8f62f5cpynngscyr5";
  };
  pkgs = import nixpkgs { config = { allowUnfree = true; }; overlays = []; };
  nixosFunc = import (pkgs.path + "/nixos");
  machine = nixosFunc {
    configuration = {
      boot.loader.grub.devices = [ "/dev/sda" ];
      fileSystems = {
        "/" = { device = "/dev/sda"; };
      };
      environment.systemPackages = with pkgs; [
        quartus-prime-lite
      ];
      imports = [
        # /path/to/an/existing/nixos/module.nix
      ];
      services.xserver = {
        enable = true;
        displayManager.gdm.enable = true;
        desktopManager.gnome.enable = true;
      };
      users.users.test = {
        isNormalUser = true;
        initialPassword = "test";
      };
    };
  };
in
  machine.vm
```

Then run `nix-build ./test.nix && ./result/bin/run-nixos-vm`, login with test/test and run Quartus.

You can use this to play around with your config. Perhaps you find what GNOME pulls in that can be added to your minimal WM setup?

**Comment by Abaan404** [comment]
Hi! Thank you so much for your help, turns out it was my nvidia drivers (maybe wayland?) not being detected by my compositor (hyprland). Infact anything x11 would segfault and I just never realized it since everything I use is on wayland these days (quartus was the first x11 app I ran but since it doesnt print an error I just never found out what went wrong).

If anyone else comes across this I simply moved my hyprland config from home-manager to `configuration.nix`.

Once again thanks alot and sorry for any inconvinience.
